### PR TITLE
Documentation collaborative learning scenario [19185]

### PR DESCRIPTION
**Pull request by irenebm**

### CODE REVIEW

File: amlip_docs/rst/user_manual/scenarios/collaborative_learning.rst [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+.. _user_manual_scenarios_collaborative_learning:
+
+###############################
+Collaborative Learning Scenario
+###############################
+
+This :term:`Scenario` performs the action of  work on the same problem independently in remote nodes,
+using different training datasets, in order to share the knowledge learned with other nodes and
+do not block any other important actions that may require to run in the same device.
+This creates a more complex and accurate model.
```
Comment (juanlofer-eprosima) [inline amlip_docs/rst/user_manual/scenarios/collaborative_learning.rst:12]:
```suggestion
This intends to lead towards a more complex and accurate model.
```

File: amlip_docs/rst/user_manual/scenarios/collaborative_learning.rst [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+It uses the :ref:`user_manual_dds_rpc` communication to publish those problems in an efficient way.
```
Comment (juanlofer-eprosima) [inline amlip_docs/rst/user_manual/scenarios/collaborative_learning.rst:13]:
```suggestion
It leverages the :ref:`user_manual_dds_rpc` communication protocol/paradigm in order to exchange all required information (model requests/replies) in an efficient way.
```

File: amlip_docs/rst/user_manual/nodes/model_manager_sender.rst [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+            // Create ModelManagerSender Node
+            eprosima::amlip::node::ModelManagerSenderNode model_sender_node(id, statistics);
+
+            // Create waiter
+            std::shared_ptr<eprosima::utils::event::BooleanWaitHandler> waiter =
+                std::make_shared<eprosima::utils::event::BooleanWaitHandler>(false, true);
+
+            // Create listener to process requests and return replies
+            std::shared_ptr<CustomModelReplier> replier =
```
Comment (juanlofer-eprosima) [inline amlip_docs/rst/user_manual/nodes/model_manager_sender.rst:49]: This implementation is missing right?

File: amlip_docs/rst/user_manual/scenarios/collaborative_learning.rst [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    A more specific Data Type will be implemented in future releases for efficiency improvements.
+
+
+.. _user_manual_scenarios_collaborative_learning_solution:
+
+Model Solution Data Type
+========================
+
+The **Solution** Data Type represents a problem reply with a more complex and accurate model.
```
Comment (juanlofer-eprosima) [inline amlip_docs/rst/user_manual/scenarios/collaborative_learning.rst:38]: Get rid of the "more complex and accurate model" everywhere please.

File: amlip_docs/rst/user_manual/scenarios/collaborative_learning.rst [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    A more specific Data Type will be implemented in future releases for efficiency improvements.
+
+
+.. _user_manual_scenarios_collaborative_learning_statistics:
+
+
+Model Statistics Data Type
+==========================
+
+The **Statistics** Data Type represents the statistics of a model, such as the weights or the dataset.
```
Comment (juanlofer-eprosima) [inline amlip_docs/rst/user_manual/scenarios/collaborative_learning.rst:53]: If multiple models is a possibility, change to plural where applies.

Comment (juanlofer-eprosima) [inline amlip_docs/rst/user_manual/scenarios/collaborative_learning.rst:53]: Weights? You mean model size? Or accuracy measures?

**Review by juanlofer-eprosima — CHANGES_REQUESTED** [review]

File: amlip_docs/rst/user_manual/scenarios/collaborative_learning.rst (hunk 5 above)
Comment (irenebm) [reply amlip_docs/rst/user_manual/scenarios/collaborative_learning.rst:53]: I mean model size, but it is just an example of what can be included, could be whatever we want.

Comment (irenebm) [reply amlip_docs/rst/user_manual/scenarios/collaborative_learning.rst:53]: I mean model size, but it is just an example of what can be included, could be whatever we want.

File: amlip_docs/rst/figures/rpc_diagram.png
[the remark this answers is not in the document]
Comment (irenebm) [reply amlip_docs/rst/figures/rpc_diagram.png:1]: No, it is the time the server is running, blocked or not.

File: amlip_docs/rst/user_manual/nodes/model_manager_receiver.rst [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+                        const std::shared_ptr<eprosima::utils::event::BooleanWaitHandler>& waiter)
+                    : waiter_(waiter)
+                {
+                    // Do nothing
+                }
+
+                virtual bool statistics_received (
+                        const eprosima::amlip::types::ModelStatisticsDataType statistics) override
+                {
+                    // Decide if we want the model based on the statistics received
```
Comment (juanlofer-eprosima) [inline amlip_docs/rst/user_manual/nodes/model_manager_receiver.rst:48]:
```suggestion
                    // Always request model
```

File: amlip_docs/rst/user_manual/nodes/model_manager_receiver.rst [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+                    return true;
+                }
+
+                virtual bool model_received (
+                        const eprosima::amlip::types::ModelSolutionDataType model) override
+                {
+                    logUser(AMLIPCPP_MANUAL_TEST, "Model received: " << model << " .");
```
Comment (juanlofer-eprosima) [inline amlip_docs/rst/user_manual/nodes/model_manager_receiver.rst:55]: This wouldn't compile I believe, and anyways the log category is not right.

**Review by juanlofer-eprosima — CHANGES_REQUESTED** [review]

**Review by juanlofer-eprosima — APPROVED** [review]
LGTM

**Pull request merged**